Mathlib/FieldTheory/Minpoly/Field.lean

```diff
@@ -118,6 +118,15 @@ theorem eq_of_irreducible_of_monic [Nontrivial B] {p : A[X]} (hp1 : Irreducible
     mul_one (minpoly A x) ▸ hq.symm ▸ Associated.mul_left _
       (associated_one_iff_isUnit.2 <| (hp1.isUnit_or_isUnit hq).resolve_left <| not_isUnit A x)
 
+theorem eq_iff_aeval_eq_zero [Nontrivial B] {p : A[X]} (hp1 : Irreducible p) (hp3 : p.Monic) :
+    p = minpoly A x ↔ Polynomial.aeval x p = 0 :=
+  ⟨(· ▸ aeval A x), (eq_of_irreducible_of_monic hp1 · hp3)⟩
+
+theorem eq_iff_aeval_minpoly_eq_zero [IsDomain B] {C} [Ring C] [Algebra A C] [Nontrivial C]
```

Comment:
```suggestion
theorem eq_iff_aeval_minpoly_eq_zero [IsDomain B] {C : Type*} [Ring C] [Algebra A C] [Nontrivial C]
```

Reply:
Is it not clear from [Ring C] that C is a Type*? Or are there other concerns? Worried that C may be inferred to be of some fixed universe rather than polymorphic?

Reply:
I think `{C}` is fine.